skyportal/handlers/api/spatial_catalog.py

```diff
@@ -0,0 +1,379 @@
+        for entry in entries:
+            tiles = [
+                SpatialCatalogEntryTile(
+                    entry_name=entry.entry_name, healpix=uniq, probdensity=probdensity
+                )
+                for uniq, probdensity in zip(entry.uniq, entry.probdensity)
+            ]
+
+            session.add_all(tiles)
+        session.commit()
+
+        flow = Flow()
+        flow.push(
+            '*',
+            "skyportal/REFRESH_SPATIAL_CATALOGS",
+        )
+
+        end = time.time()
+        duration = end - start
+
+        log(f"Generated catalog with ID {catalog_id} in {duration} seconds")
+    except Exception as e:
+        log(f"Unable to generate catalog: {e}")
+    finally:
+        Session.remove()
+
+
+def delete_catalog(catalog_id):
+
+    log(f"Deleting catalog with ID {catalog_id}")
+
+    session = Session()
+    try:
+        catalog = session.scalar(
+            sa.select(SpatialCatalog).where(SpatialCatalog.id == int(catalog_id))
+        )
+        session.delete(catalog)
+        session.commit()
+
+        flow = Flow()
+        flow.push(
+            '*',
+            "skyportal/REFRESH_SPATIAL_CATALOGS",
+        )
+
+        log(f"Deleted catalog with ID {catalog_id}")
+    except Exception as e:
+        log(f"Unable to delete catalog: {e}")
+    finally:
+        Session.remove()
+
+
+class SpatialCatalogHandler(BaseHandler):
+    @auth_or_token
+    def post(self):
+        """
+        ---
+        description: Ingest a Spatial Catalog
+        tags:
+          - galaxies
+        requestBody:
+          content:
+            application/json:
+              schema: SpatialCatalogHandlerPost
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            id:
+                              type: integer
+                              description: New spatial catalog ID
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        data = self.get_json()
+        catalog_name = data.get('catalog_name')
+        catalog_data = data.get('catalog_data')
+
+        if catalog_name is None:
+            return self.error("catalog_name is a required parameter.")
+        if catalog_data is None:
+            return self.error("catalog_data is a required parameter.")
+
+        # check RA bounds
+        if any([(x < 0) or (x >= 360) for x in catalog_data['ra']]):
+            return self.error("ra should span 0=<ra<360.")
+
+        # check Declination bounds
+        if any([(x > 90) or (x < -90) for x in catalog_data['dec']]):
+            return self.error("declination should span -90<dec<90.")
```

Comment:
careful here, you might want to first check if `ra` and `dec` are in the dict to avoid an error

Reply:
yes.